src/backend/distributed/commands/schema.c

```diff
@@ -109,6 +123,17 @@ PreprocessDropSchemaStmt(Node *node, const char *queryString,
 
 	String *schemaVal = NULL;
 	foreach_ptr(schemaVal, distributedSchemas)
+	{
+		bool missingOk = false;
+		Oid schemaId = get_namespace_oid(strVal(schemaVal), missingOk);
+		if (IsTenantSchema(schemaId))
+		{
+			UnregisterTenantSchema(schemaId);
```

Comment:
We need to delete pg_dist_colocation entry too. If the schema has at least one tenant table, we already do so via master_remove_partition_metadata(). However, this is not the case for an empty schema.

Reply:
oh, I don't think so